app/ts/components/pages/WebsiteAccess.tsx

```diff
@@ -0,0 +1,561 @@
+				)) }
+			</div>
+		</Collapsible>
+	)
+}
+
+const AddressAccessCard = ({ website, addressAccess }: { website: Website, addressAccess: WebsiteAddressAccess }) => {
+	const setAddressAccess = (event: Event) => {
+		if (!(event.target instanceof HTMLInputElement)) return
+		sendPopupMessageToBackgroundPage({ method: 'popup_allowOrPreventAddressAccessForWebsite', data: { website, address: addressAccess.address, allowAccess: event.target.checked } })
+	}
+
+	return (
+		<div style = { { display: 'grid', gridTemplateColumns: 'minmax(0,1fr) min-content min-content', columnGap: '1rem', alignItems: 'center' } }>
+			<AddressCard address = { addressAccess.address } />
+			<RemoveAddressConfirmation websiteOrigin = { website.websiteOrigin } address = { addressAccess.address } />
+			<Switch checked = { addressAccess.access } onChange = { setAddressAccess } />
+		</div>
+	)
+}
+
+const RemoveAddressConfirmation = ({ websiteOrigin, address }: { address: bigint, websiteOrigin: string }) => {
+	const { searchQuery } = useWebsiteAccess()
+	const addressString = serialize(EthereumAddress, address)
+
+	const removeAddressAccessForWebsite = async () => {
+		await updateWebsiteAccess((existingAccessList) => {
+			const newWebsiteAccessMap = new Map(existingAccessList.map(access => [access.website.websiteOrigin, { ...access }]))
+			if (newWebsiteAccessMap.has(websiteOrigin)) {
+				const websiteAccessOfWebsiteOrigin = newWebsiteAccessMap.get(websiteOrigin)!
+				if (!websiteAccessOfWebsiteOrigin.addressAccess) return existingAccessList
+				const addressAccessOfWebsiteOriginMap = new Map(websiteAccessOfWebsiteOrigin.addressAccess.map(addressAccess => [addressAccess.address, { ...addressAccess }]))
+				if (addressAccessOfWebsiteOriginMap.has(address)) addressAccessOfWebsiteOriginMap.delete(address)
+				websiteAccessOfWebsiteOrigin.addressAccess = Array.from(addressAccessOfWebsiteOriginMap.values())
+			}
+			return Array.from(newWebsiteAccessMap.values())
+		})
+
+		sendPopupMessageToBackgroundPage({ method: 'popup_retrieveWebsiteAccess',  data: { query: searchQuery.value } })
+	}
+
+	const confirmOrRejectRemoval = (returnValue: string) => {
+		if (returnValue !== 'confirm') return
+		removeAddressAccessForWebsite()
+	}
+
+	return (
+		<Modal>
+			<Modal.Open class = 'btn btn--ghost'><TrashIcon /></Modal.Open>
+			<Modal.Dialog class = 'dialog' style = { { textAlign: 'center', color: 'var(--disabled-text-color)' } } onClose = { confirmOrRejectRemoval }>
+				<h2 style = { { fontWeight: 600, fontSize: '1.125rem', color: 'var(--text-color)', marginBlock: '1rem' } }>Removing Address</h2>
+				<p style = { { marginBlock: '0.5rem' } }>This will prevent <pre>{websiteOrigin}</pre> from accessing to the following address <data value = { addressString } style = { { backgroundColor: 'var(--card-bg-color)', color: 'var(--text-color)', padding: '2px 4px', borderRadius: 2 } }>{ addressString }</data></p>
```

Comment:
```suggestion
				<p style = { { marginBlock: '0.5rem' } }>This will prevent <pre>{ websiteOrigin }</pre> from accessing to the following address <data value = { addressString } style = { { backgroundColor: 'var(--card-bg-color)', color: 'var(--text-color)', padding: '2px 4px', borderRadius: 2 } }>{ addressString }</data></p>
```

Reply:
we should also show the address name & icon here instead of the address number